### PR TITLE
[Docs site] Rss feed test

### PR DESCRIPTION
**Pull request by kodster28**
Provides an MVP solution for #6048

### CODE REVIEW

**Comment by kodster28** [comment]
Thoughts on this, @WalshyDev and @GregBrimble?

From what I can tell about Hugo's default behavior, page-level RSS feeds aren't really supported (more meant to be when there are new posts on a blog). So really this only works for pages organized like https://rss-feed-test.cloudflare-docs-7ou.pages.dev/waf/change-log/, which add a new page for each change that'll then be picked up in the RSS feed.

Not perfect by any means and will require restructuring our current changelogs to match... but think it provides a way forward w/o a ton of custom development.

For visibility, adding an RSS icon to pages that have an RSS feed enabled.
Screenshot 2022-12-16 at 10 57 42 AM

Also adding a top-level page at https://rss-feed-test.cloudflare-docs-7ou.pages.dev/rss/ to list all the available options.

**Comment by kodster28** [comment]
Doing a touch more digging, think that the structure of their [.GitInfo](https://gohugo.io/variables/git/#lastmod) variable excludes the possibility of an easy soln for page-level changes. It only stores the last commit info per a page, probably based off the use case of `.LastMod`.

**Comment by kodster28** [comment]
Actually... maybe I'm overcomplicating this... it'd probably be easier just to use GitHub's built-in RSS feed per a file or folder.

https://github.com/cloudflare/cloudflare-docs/commits/production/content/waf/change-log.atom
https://github.com/cloudflare/cloudflare-docs/commits/production/content/stream/changelog.md.atom

This seems like the least amount of work + the most flexibility.

File: layouts/_default/rss.xml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    {{ printf "<atom:link href=%q rel=\"self\" type=%q />" .Permalink .MediaType | safeHTML }}
+    {{- end -}}
+    {{ range $pages }}
+    <item>
+      <title>{{ .Title }}</title>
+      <link>{{ .Permalink }}</link>
+      <pubDate>{{ .Lastmod.Format "Mon, 02 Jan 2006 15:04:05 -0700" | safeHTML }}</pubDate>
+      {{ with .Site.Author.email }}<author>{{.}}{{ with $.Site.Author.name }} ({{.}}){{end}}</author>{{end}}
+      <guid>{{ .Permalink }}</guid>
+      <description>{{ .Summary | html }}</description>
```
Comment (pedrosousa) [inline layouts/_default/rss.xml]: While this description seems to contain relevant information like CVE numbers, the format is not great: it's essentially a bunch of table cells bundled together into a sentence. 
Maybe we could require a useful summary sentence in the Markdown front matter for these changelog pages, which could include information like CVE numbers or affected rule IDs?

**Comment by kodster28** [comment]
When you specify an `rss: folder` or `rss: file` property in the frontmatter of a page, that page gets:

-  an RSS icon generated in the top toolbar pointing to the [GH RSS feed](https://github.com/cloudflare/cloudflare-docs/commits/production/content/waf/change-log.atom)
- a link added to the HTML head to the same feed
- Added to the [RSS feeds page](https://rss-feed-test.cloudflare-docs-7ou.pages.dev/rss/), which is also linked in the top toolbar of home pages for visibility

There's an edge case where this logic will break if you specify `rss: file` on `_index.md` pages, but a) we can document it and b) feel like that's not an onerous burden on anyone.

Added examples for [WAF](https://rss-feed-test.cloudflare-docs-7ou.pages.dev/waf/change-log/) -- which is a folder -- and [Stream](https://rss-feed-test.cloudflare-docs-7ou.pages.dev/stream/changelog/) -- which is a file.

TBD --> Set up standards for changelog commits so they're more easily identifiable in the logs

File: layouts/partials/topbar.tools.html [hunk 2]
```diff
@@ -10,12 +14,47 @@
...
+        <a class="Link Link-without-underline RSS" href="{{$rssFeed}}" target="_blank">
+          <svg viewBox="0 0 485 560" role="img" aria-labelledby="title-1234" xmlns="http://www.w3.org/2000/svg">
+            <title id="title-1234">RSS icon</title>
+            <path d="M0 64C0 46.3 14.3 32 32 32c229.8 0 416 186.2 416 416c0 17.7-14.3 32-32 32s-32-14.3-32-32C384 253.6 226.4 96 32 96C14.3 96 0 81.7 0 64zM128 416c0 35.3-28.7 64-64 64s-64-28.7-64-64s28.7-64 64-64s64 28.7 64 64zM32 160c159.1 0 288 128.9 288 288c0 17.7-14.3 32-32 32s-32-14.3-32-32c0-123.7-100.3-224-224-224c-17.7 0-32-14.3-32-32s14.3-32 32-32z"></path>
+          </svg>
+        </a>
+      </div>
+
+      <span class="Tooltip" role="tooltip" position="bottom-end">
+        View RSS Feed
```
Comment (pedrosousa) [inline layouts/partials/topbar.tools.html]: To manage expectations, how about using "View GitHub RSS feed" instead? Would that be too confusing?
My main issue with not mentioning GitHub is that all feed entry links will point to GH pull requests.

File: layouts/partials/rss-icon.html [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<a class="Button DocsMarkdown--resource" href="/rss">
```
Comment (pedrosousa) [inline layouts/partials/rss-icon.html]: Are we using this file? Also, the `rss.svg` file doesn't seem to exist.

File: layouts/shortcodes/rss-list.html [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+{{ range .Site.Pages }} 
+{{- $data := (index $.Site.Data .Section) -}}
+{{- $title := default .Title .Params.meta.title  -}}
+{{- $rssPath := "https://github.com/cloudflare/cloudflare-docs/commits/production/content" -}}
+{{- $contentPath :=  replaceRE `\/$` "" .RelPermalink -}}
+{{- $folderPlaceholder := "" -}}
+{{ $rssFeed := "" }}
+{{ with .Params.rss }}
+
+  {{ if eq . "folder" }}
```
Comment (pedrosousa) [inline layouts/shortcodes/rss-list.html]: Not a serious issue, but we're not accounting for invalid `rss` values in frontmatter (different from `folder` and `file`). In this case, we would create a link that wouldn't point to an RSS file.

**Review by pedrosousa — APPROVED** [review]

File: layouts/shortcodes/rss-list.html (hunk 4 above)
Comment (kodster28) [reply layouts/shortcodes/rss-list.html]: Should be fixed in the latest commit, both for the icon + the link the `<head>`
